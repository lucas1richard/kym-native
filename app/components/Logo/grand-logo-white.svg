<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 201.47 61"><defs><style>.cls-1{fill:#ff8d8d;}.cls-2{fill:#ffe15c;}.cls-3{fill:#4ecbc2;}.cls-4{fill:#ff4949;}.cls-5{fill:#ecf9f8;}.cls-6{fill:#2d9890;}.cls-7{fill:none;stroke:#174460;stroke-linecap:round;stroke-linejoin:round;stroke-width:5px;}.cls-8{fill:#ffffff;}</style></defs><title>grand_logo</title><polygon class="cls-1" points="2.5 16.5 26.75 30.5 26.75 2.39 2.5 16.5"/><polygon class="cls-2" points="26.75 30.5 2.5 44.5 2.5 16.5 26.75 30.5"/><polygon class="cls-3" points="26.75 30.5 2.5 44.5 26.75 58.5 26.75 30.5"/><polygon class="cls-4" points="51 16.5 26.75 30.5 26.75 2.39 51 16.5"/><polygon class="cls-5" points="26.75 30.5 51 44.5 51 16.5 26.75 30.5"/><polygon class="cls-6" points="26.75 30.5 51 44.5 26.75 58.5 26.75 30.5"/><polygon class="cls-7" points="51 44.5 51 16.5 26.75 2.5 2.5 16.5 2.5 44.5 26.75 58.5 51 44.5"/><path class="cls-8" d="M84.29,42.84c-7,0-12.78-4.9-12.78-12.45S77.27,18,84.29,18a11.2,11.2,0,0,1,8.53,3.53A2.34,2.34,0,0,1,93.44,23,2.42,2.42,0,0,1,91,25.35a2.49,2.49,0,0,1-1.73-.72,7.25,7.25,0,0,0-5-2.12c-4.36,0-7.52,3.35-7.52,7.88s3.17,7.88,7.52,7.88a8.1,8.1,0,0,0,5.11-1.84V33.56H85a2.23,2.23,0,0,1,0-4.46h6.59A2.89,2.89,0,0,1,94.52,32V36C94.52,40.4,89.15,42.84,84.29,42.84Z"/><path class="cls-8" d="M114.61,33.23l3.31,5.62a2.59,2.59,0,0,1,.32,1.22,2.55,2.55,0,0,1-2.52,2.52,2.5,2.5,0,0,1-2.2-1.3l-4.1-7.45h-3.74v6.23a2.56,2.56,0,0,1-5.11,0V21.25a2.89,2.89,0,0,1,2.88-2.88h8.35c5,0,8.06,3.28,8.06,7.74A6.89,6.89,0,0,1,114.61,33.23Zm-8.93-10.37v6.48H111c2.05,0,3.6-1.22,3.6-3.24s-1.55-3.24-3.6-3.24Z"/><path class="cls-8" d="M143.66,42.59a2.65,2.65,0,0,1-2.48-1.69l-.94-2.59H129.94l-1,2.59a2.66,2.66,0,0,1-2.52,1.69,2.72,2.72,0,0,1-2.7-2.74,2.32,2.32,0,0,1,.18-1l7-18a4.25,4.25,0,0,1,3.92-2.7,4.35,4.35,0,0,1,4.36,2.7l7,18a3.11,3.11,0,0,1,.18,1A2.75,2.75,0,0,1,143.66,42.59Zm-8.57-19.11-3.74,10.33h7.49Z"/><path class="cls-8" d="M169.54,42.59A3.2,3.2,0,0,1,167,41.3L156.33,26.72V40.07a2.56,2.56,0,0,1-5.11,0V21.43a3.35,3.35,0,0,1,3.35-3.28,3.48,3.48,0,0,1,2.81,1.4L167.6,33.45V20.67a2.56,2.56,0,0,1,5.11,0V39.5A3.13,3.13,0,0,1,169.54,42.59Z"/><path class="cls-8" d="M188.73,42.38h-6.59a2.89,2.89,0,0,1-2.88-2.88V21.25a2.89,2.89,0,0,1,2.88-2.88h6.59c7.52,0,12.74,4.79,12.74,12S196.25,42.38,188.73,42.38Zm0-19.51h-4.36v15h4.36a7.19,7.19,0,0,0,7.52-7.49C196.25,26.14,193.66,22.87,188.73,22.87Z"/></svg>
