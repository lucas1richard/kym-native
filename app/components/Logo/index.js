/**
*
* VerticalLogo
*
*/
import React from 'react';
import Icon from 'components/Icon';
import { StyleSheet, View } from 'react-native';

const height = 60;
const width = height * (241.47 / 61);

function LogoComponent() {
  return (
    <View style={styles.logo}>
      <Icon name="grandLogo" height={height} width={width} viewBox="0 0 201.47 61" />
    </View>
  );
}

export default LogoComponent;

const styles = StyleSheet.create({
  logo: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  }
});
