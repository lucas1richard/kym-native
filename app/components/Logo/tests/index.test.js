import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Logo from '../index';

describe('<Logo />', () => {
  const renderedComponent = (props = {}) => shallow(<Logo {...props} />);

  it('should render without blowing up', () => {
    expect(toJson(renderedComponent())).toMatchSnapshot();

    const props = {
      white: true
    };
    expect(toJson(renderedComponent(props))).toMatchSnapshot();
  });
});
