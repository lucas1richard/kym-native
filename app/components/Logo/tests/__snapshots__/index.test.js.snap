// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render without blowing up 1`] = `
<div>
  <Styled(Img)
    alt="Grand"
    src="IMAGE_MOCK"
  />
  <Styled(Img)
    alt="Grand"
    src="IMAGE_MOCK"
  />
</div>
`;

exports[`<Logo /> should render without blowing up 2`] = `
<div>
  <Styled(Img)
    alt="Grand"
    src="IMAGE_MOCK"
  />
  <Styled(Img)
    alt="Grand"
    src="IMAGE_MOCK"
  />
</div>
`;
